Bottom: 4ff61e9c5e98a823f736525f19acf93f75e99d2f
Top:    320ae0db90b98660cc443980e88fcb576f39d8b7
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-02-16 18:57:18 +0800

Refresh of virtio-net-optimize-small

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 857054c..32c1e84 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -813,6 +813,7 @@ static int add_recvbuf_small(struct virtnet_info *vi, struct receive_queue *rq,
 
 	buf = (char *)page_address(alloc_frag->page) + alloc_frag->offset;
 	buf += NET_IP_ALIGN + xdp_headroom;
+	get_page(alloc_frag->page);
 	sg_init_one(rq->sg, buf, GOOD_PACKET_LEN);
 	err = virtqueue_add_inbuf(rq->vq, rq->sg, 1, buf, gfp);
 	if (err < 0)
